setup.py

```diff
@@ -0,0 +1,201 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from distutils.spawn import find_executable
+from distutils import sysconfig, log
+import setuptools
+import setuptools.command.build_py
+import setuptools.command.develop
+import setuptools.command.build_ext
+
+from collections import namedtuple
+import multiprocessing
+import os
+import shlex
+import subprocess
+import sys
+from textwrap import dedent
+
+TOP_DIR = os.path.realpath(os.path.dirname(__file__))
+SRC_DIR = os.path.join(TOP_DIR, 'caffe2')
+PROTOC = find_executable('protoc')
+
+install_requires = set()
+setup_requires = set()
+test_requires = set()
+
+################################################################################
+# Pre Check
+################################################################################
+
+assert find_executable('cmake'), 'Could not find "cmake" executable!'
+assert find_executable('make'), 'Could not find "make" executable!'
+
+################################################################################
+# Version
+################################################################################
+
+try:
+    git_version = subprocess.check_output(['git', 'rev-parse', 'HEAD'],
+                                          cwd=TOP_DIR).decode('ascii').strip()
+except subprocess.CalledProcessError:
+    git_version = None
+
+with open(os.path.join(TOP_DIR, 'VERSION_NUMBER')) as version_file:
+    VersionInfo = namedtuple('VersionInfo', ['version', 'git_version'])(
+        version=version_file.read().strip(),
+        git_version=git_version
+    )
```

Comment:
Instead of a call to git and a VERSION_FILE, you can use just a single git command to get a descriptive version. Then the VERSION_FILE can be removed.

```
$ git describe --tags HEAD
v0.8.1-687-g08574bb
```

Reply:
We used to do that in onnx, but later change to this way because source distribution, it's not a git repo. Also it's helpful for doing importing.

Reply:
I see, makes sense. Perhaps we can use the git way if it's a git repo and fall back to VERSION if it doesn't work? It's nice to have both the tag + distance + sha IMO.

Reply:
We can add as many entries as we want in the VersionInfo (say git_describe=$(git describe --tags)) :-)